### PR TITLE
[Travis] update phan version to 3.0 and fix the phan issues

### PR DESCRIPTION
**Pull request by kongtiaowang**
## Brief summary of changes

 fix phan and composer issue

#### Testing instructions (if applicable)

passing Travis

### CODE REVIEW

**Comment by spell00** [comment]
@driusan This is blocking the PRs I need to merge this week, moving this forward as soon as possible would be very helpful

File: .phan/config.php
Comment (driusan) [inline .phan/config.php:41]: Why are these re-ignored? A lot of work went into fixing each of the phan rules that aren't ignored and Travis should have prevented any regressions from happening.

File: modules/imaging_browser/php/imagingbrowserrowprovisioner.class.inc
Comment (driusan) [inline modules/imaging_browser/php/imagingbrowserrowprovisioner.class.inc:70]: If it's unused it should be deleted, not commented out.. @cmadjar do you know why this would be here but never used? I don't see any references to the variable in the code when I ctrl-f in the file on GitHub.

File: php/libraries/NDB_BVL_Battery.class.inc
Comment (driusan) [inline php/libraries/NDB_BVL_Battery.class.inc:248]: These assertions should not be removed.

File: php/libraries/NDB_BVL_Battery.class.inc
Comment (driusan) [inline php/libraries/NDB_BVL_Battery.class.inc:549]: Nor these

File: php/libraries/NDB_Form.class.inc
Comment (driusan) [inline php/libraries/NDB_Form.class.inc:87]: These function calls should not be removed.

**Review by driusan — CHANGES_REQUESTED** [review]

**Comment by christinerogers** [comment]
hi @kongtiaowang is this ready for @driusan 's re-review? 
 trying to unblock some downstream PRs that are critical this week...

File: php/libraries/NDB_BVL_Battery.class.inc
Comment (driusan) [inline php/libraries/NDB_BVL_Battery.class.inc:544]: We shouldn't be suppressing warnings, just don't assign the function call to a variable.

File: php/libraries/NDB_BVL_Instrument_LINST.class.inc
Comment (driusan) [inline php/libraries/NDB_BVL_Instrument_LINST.class.inc:202]: This is changing the semantics of the function quite significantly. What are you trying to solve?

File: php/libraries/NDB_Form.class.inc
Comment (driusan) [inline php/libraries/NDB_Form.class.inc:57]: Don't suppress warnings.

**Review by driusan — CHANGES_REQUESTED** [review]

**Comment by driusan** [comment]
@kongtiaowang can you prioritize this? I didn't realize when we were doing the assignments yesterday that this is preventing us from upgrading composer to 2.0, which might be required to look into alternatives to Travis.

File: php/libraries/NDB_BVL_Battery.class.inc
Comment (kongtiaowang) [reply php/libraries/NDB_BVL_Battery.class.inc:544]: Phan error for unused variable.

File: php/libraries/NDB_Form.class.inc
Comment (kongtiaowang) [reply php/libraries/NDB_Form.class.inc:57]: Phan error for unused variable.

File: php/libraries/NDB_BVL_Instrument_LINST.class.inc
Comment (kongtiaowang) [reply php/libraries/NDB_BVL_Instrument_LINST.class.inc:202]: php/libraries/NDB_BVL_Instrument_LINST.class.inc:200 PhanTypeMismatchDimFetch When fetching an array index from a value of type list<mixed>|string, found an array index of type 'rules', but expected the index to be of type int
php/libraries/NDB_BVL_Instrument_LINST.class.inc:221 PhanTypeMismatchDimFetch When fetching an array index from a value of type list<mixed>|string, found an array index of type 'group', but expected the index to be of type int
php/libraries/NDB_BVL_Instrument_LINST.class.inc:222 PhanTypeMismatchDimFetch When fetching an array index from a value of type list<mixed>|string, found an array index of type 'group', but expected the index to be of type int
php/libraries/NDB_BVL_Instrument_LINST.class.inc:225 PhanTypeMismatchDimFetch When fetching an array index from a value of type list<mixed>|string, found an array index of type 'message', but expected the index to be of type int
[debug] Restarted process exited 1

File: php/libraries/NDB_BVL_Battery.class.inc
Comment (driusan) [reply php/libraries/NDB_BVL_Battery.class.inc:544]: The solution is to not declare unused variables, not suppress the error.

File: php/libraries/NDB_BVL_Instrument_LINST.class.inc
Comment (driusan) [reply php/libraries/NDB_BVL_Instrument_LINST.class.inc:202]: from what I can tell phan isn't inferring the type of `$this->XINRules[$elname]` properly based on the format it's generated in in XINRegisterRule. Your fix will satify phan but break the rules themselves. 

I managed to fix phan without changing the logic here by telling it that it's an array (since it seems to always be an array in XINRegisterRule)

```
$registered_rules = $this->XINRules[$elname];
'@phan-var array $registered_rules';
```

File: .github/workflows/docker-image.yml
Comment (driusan) [inline .github/workflows/docker-image.yml:16]: These files should not be in this PR.

File: php/libraries/NDB_BVL_Battery.class.inc
Comment (driusan) [inline php/libraries/NDB_BVL_Battery.class.inc:242]: This should still not be suppressed

**Review by driusan — CHANGES_REQUESTED** [review]

File: php/libraries/NDB_BVL_Battery.class.inc
Comment (driusan) [inline php/libraries/NDB_BVL_Battery.class.inc:248]: Why are you doing this? If the error is "unused variable" assigning the variable a type won't fix it. You need to just not assign the function call to anything like

```
$this->_assertBatterySelected()
```

instead of

```
$error = $this->_assertBatterySelected();
```

**Review by driusan — APPROVED** [review]
needs Travis to pass before merging, but the code looks good now.

**Pull request merged**
